src/OnboardingSPA/components/SocialMediaForm/socialMediaModal.js

```diff
@@ -0,0 +1,63 @@
+import { __ } from '@wordpress/i18n';
+import { useNavigate } from 'react-router-dom';
+import { Fragment } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { Button, ButtonGroup, Modal } from '@wordpress/components';
+
+import { store as nfdOnboardingStore } from '../../store';
+
+const SocialMediaModal = ( {
+	showModal,
+	setShowModal,
+	modalTitle,
+	modalText,
+} ) => {
+	const navigate = useNavigate();
+	const { navErrorModalPath } = useSelect( ( select ) => {
+		return {
+			navErrorModalPath:
+				select( nfdOnboardingStore ).getNavErrorModalPath(),
+		};
+	}, [] );
+
+	const { addNavErrorModalCode, resetNavErrorModal } =
+		useDispatch( nfdOnboardingStore );
+
+	const closeModal = () => {
+		resetNavErrorModal();
+		addNavErrorModalCode( 400 );
+		setShowModal( false );
```

Comment:
Why are we resetting and then adding a `400` error code and then closing the modal?